File: tiktok_backend/models/videoModel.js
// models/video.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const VideoSchema = new Schema({
    userId : {type : String } , 
    name: { type: String, required: true },
    filePath: { type: String, required: true },
    createdAt: { type: Date, default: Date.now },
});

const Video = mongoose.model('Video', VideoSchema);

module.exports = Video;

File: tiktok_app/lib/models/video_model.dart
class Video {
  final String? id;
  final String? userId;
  final String? userEmail; // Optional field, as it is not present in all objects
  final String? name;
  final String? filePath;
  final DateTime? createdAt;
  final String? image ; 

  Video({
    required this.id,
    required this.userId,
    this.userEmail,
    required this.name,
    required this.filePath,
    required this.createdAt,
     this.image
  });

  factory Video.fromJson(Map<String, dynamic> json) {
    return Video(
      id: json['_id'],
      userId: json['userId'],
      userEmail: json['userEmail'],
      name: json['name'],
      filePath: json['filePath'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
}